pyt/__main__.py

```diff
@@ -39,66 +57,66 @@ def main(command_line_args=sys.argv[1:]):  # noqa: C901
     elif args.trim_reassigned_in:
         ui_mode = UImode.TRIM
 
-    path = os.path.normpath(args.filepath)
-
-    if args.ignore_nosec:
-        nosec_lines = set()
-    else:
-        file = open(path, 'r')
-        lines = file.readlines()
-        nosec_lines = set(
-            lineno for
-            (lineno, line) in enumerate(lines, start=1)
-            if '#nosec' in line or '# nosec' in line
-        )
-
-    if args.project_root:
-        directory = os.path.normpath(args.project_root)
-    else:
-        directory = os.path.dirname(path)
-    project_modules = get_modules(directory)
-    local_modules = get_directory_modules(directory)
-
-    tree = generate_ast(path)
-
-    cfg = make_cfg(
-        tree,
-        project_modules,
-        local_modules,
-        path
-    )
-    cfg_list = [cfg]
-    framework_route_criteria = is_flask_route_function
-    if args.adaptor:
-        if args.adaptor.lower().startswith('e'):
-            framework_route_criteria = is_function
-        elif args.adaptor.lower().startswith('p'):
-            framework_route_criteria = is_function_without_leading_
-        elif args.adaptor.lower().startswith('d'):
-            framework_route_criteria = is_django_view_function
-    # Add all the route functions to the cfg_list
-    FrameworkAdaptor(
-        cfg_list,
-        project_modules,
-        local_modules,
-        framework_route_criteria
+    files = discover_files(
+        args.targets,
+        args.excluded_paths,
+        args.recursive
     )
+    vulnerabilities = list()
```

Comment:
i created list before loop as you said. the same problem about last item is still continue.
![a](https://user-images.githubusercontent.com/19760132/41499890-360be654-7191-11e8-8dc9-f47272106a15.png)

Reply:
Hmm, That's odd, I'll checkout/test your code later today to try and find the issue. 👍

Reply:
That's really weird, I'll look more in-depth on Monday 👍

Reply:
I checked out your branch and it was `append` vs. `extend`